### ISSUE TITLE
vcat error for appending scalars

### ISSUE THREAD

**Issue by ChrisRackauckas** [FluxML/Zygote.jl#538]
MWE on master:

```julia
using Zygote
u0 = ones(10); dx =0.1
Zygote.gradient(u0) do u
    sum([zero(eltype(u)); (u[3:end] - 2.0.*u[2:end-1] + u[1:end-2]) ./ (dx^2); zero(eltype(u))])
end
```

**Comment by ChrisRackauckas** [comment]
workaround is to avoid scalars:

```julia
Zygote.gradient(u0) do u
    sum([zero(eltype(u)); (u[3:end] - 2.0.*u[2:end-1] + u[1:end-2]) ./ (dx^2); zero(eltype(u))])
end
```

**Comment by mcabbott** [comment]
Fixed by the fix for #440, I think. Which needs to become a PR.
